### PR TITLE
Port #839 forward

### PR DESCRIPTION
**Pull request by chuckha**
Signed-off-by: Chuck Ha <chuckh@vmware.com>

**What this PR does / why we need it**:
This PR moves the logic from #839 into v1alpha2

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Related to #913 

**Special notes for your reviewer**:

/assign @detiber @ncdc 

**Release note**:
```release-note
Bring additional events from v1alpha1 into v1alpha2
```

### CODE REVIEW

File: pkg/cloud/services/ec2/instances.go [hunk 1]
```diff
@@ -181,10 +181,13 @@ func (s *Service) CreateInstance(scope *scope.MachineScope) (*v1alpha2.Instance,
 	s.scope.V(2).Info("Running instance", "machine-role", scope.Role())
 	out, err := s.runInstance(scope.Role(), input)
 	if err != nil {
+		if !awserrors.IsFailedDependency(errors.Cause(err)) {
```
Comment (detiber) [inline pkg/cloud/services/ec2/instances.go]: Do we want to also bring the comment over for why we are only conditionally emitting the event.

Comment (chuckha) [reply pkg/cloud/services/ec2/instances.go]: yes we do. This bit was slightly tricky due to the v1a2 differences and I must have missed it, good catch

**Comment by ncdc** [comment]
I think we also need to remove instances.go:233:
```go
record.Eventf(s.scope.Cluster, "DeletedInstance", "Terminated instance %q", instanceID)
```

**Comment by chuckha** [comment]
oh yes, let me go through the deletes again to make sure i don't miss any other deletes

**Comment by chuckha** [comment]
updated.

**Pull request merged**